### ISSUE TITLE
Can't open /etc/easy-rsa/pki/index.txt.attr for reading, No such file or directory

### ISSUE THREAD

**Issue by graysky2** [OpenVPN/easy-rsa#132]
I get errors when I attempt to sign a new request for the first time.  Subsequent requests are signed without the error.  In other words, I have to sign 3 requests with my CA.  The first throws errors, but the next 2 work fine.

Example:
```
easyrsa import-req /mnt/mini/etc/easy-rsa/pki/reqs/ovpn-lxc.req ovpn-lxc
easyrsa sign-req server ovpn-lxc
You are about to sign the following certificate.
Please check over the details shown below for accuracy. Note that this request
has not been cryptographically verified. Please be sure it came from a trusted
source or that you have verified the request checksum with the sender.

Request subject, to be signed as a server certificate for 3650 days:

subject=
    commonName                = ovpn-lxc


Type the word 'yes' to continue, or any other input to abort.
  Confirm request details: yes
Using configuration from /etc/easy-rsa/openssl-1.0.cnf
Enter pass phrase for /etc/easy-rsa/pki/private/ca.key:
Can't open /etc/easy-rsa/pki/index.txt.attr for reading, No such file or directory
140716215046016:error:02001002:system library:fopen:No such file or directory:crypto/bio/bss_file.c:74:fopen('/etc/easy-rsa/pki/index.txt.attr','r')
140716215046016:error:2006D080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:81:
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'ovpn-lxc'
Certificate is to be certified until Jun 23 17:19:50 2027 GMT (3650 days)

Write out database with 1 new entries
Data Base Updated

Certificate created at: /etc/easy-rsa/pki/issued/ovpn-lxc.crt
```

If I repeat with other requests, they do not give the errors.
Arch Linux using easy-rsa 3.0.1-1 and openssl 1.1.0.f-1.

**Comment by danhunsaker** [comment]
Have you tested the cert produced during the run with this error output? I mean, the error certainly looks ugly, but it's not actually stopping the process — it's trying to see which sequential ID to assign to the cert when it records it, but since the file doesn't exist yet because no other certs have been generated, it can't actually open it, so it knows to use the first ID.

**Comment by graysky2** [comment]
@danhunsaker - I am experiencing show-stopping issues currently with my distro's openssl but when I get them solved I will test this.

Issues: https://bbs.archlinux.org/viewtopic.php?pid=1720537

**Comment by ecrist** [comment]
This is similar to #138 and involves OpenSSL 1.1x.  We will support this in 3.2.

**Comment by senorsmile** [comment]
This is affecting me on a new install as well.

**Comment by senorsmile** [comment]
I followed issue https://github.com/OpenVPN/easy-rsa/issues/138
and checked out the commit: 
```git checkout https://github.com/uwehermann/easy-rsa/commit/a138c0d83b0ff1feed385c5d2d7a1c25422fe04d```
this seems to fix things for now.  I assume that'll at least get merged to master some time soon?

**Comment by ecrist** [comment]
Please note, git master is NOT production, and may be broken at any time.  It is suggested you use either a release branch, or the released packages available on the github website.

This will be resolved as time permits.

Eric

**Comment by ecrist** [comment]
Wait, I just dug into this a bit further.  Take a look at line 584 from https://github.com/OpenVPN/easy-rsa/blob/master/easyrsa3/easyrsa <https://github.com/OpenVPN/easy-rsa/blob/master/easyrsa3/easyrsa>.  The patch was applied back in August.

Eric

**Comment by tomkraw1** [comment]
I've noticed that error on Windows 7 x64 with OpenVPN 2.4.6 during CA cert creation with Easy-rsa. Keys work correctly.

**Comment by aetr3yu** [comment]
Hi, just a heads up. This issue came up today as i was generating new set of certs.
**index.txt.attr**   only shows up after the **"./easyrsa build-server-full..."** command
and it's value is "unique subject = no" when it's supposed to be yes.
Im on Debian / jessie.

**Comment by ecrist** [comment]
The unique subject was changed in a recent commit. The other is just a warning and was missed in v3.0.6. We will fix it in v3.0.7.

**Comment by aetr3yu** [comment]
Thank you Sir.
